### PR TITLE
Fix parquet clippy lints (#1254)

### LINKED ISSUE #1254: Fix all clippy lints in parquet crate
**Describe the bug**
Due to "historical reasons" there are several clippy lints that are disabled in the parquet crate
https://github.com/apache/arrow-rs/blob/master/parquet/src/lib.rs#L18-L36

```rust
#![allow(incomplete_features)]
#![allow(dead_code)]
#![allow(non_camel_case_types)]
#![allow(
    clippy::approx_constant,
    clippy::cast_ptr_alignment,
    clippy::float_cmp,
    clippy::float_equality_without_abs,
    clippy::from_over_into,
    clippy::many_single_char_names,
    clippy::needless_range_loop,
    clippy::new_without_default,
    clippy::or_fun_call,
    clippy::same_item_push,
    clippy::too_many_arguments,
    clippy::transmute_ptr_to_ptr,
    clippy::upper_case_acronyms,
    clippy::vec_init_then_push
)]
```

It would be great to clean up the code to pass these lints for tidiness

**To Reproduce**
Remove one of the `#[allow]` lines above, run `clippy`

**Expected behavior**
Clippy runs cleanly without blank `#allow` across the whole crate

**Additional context**

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #1254

# Rationale for this change
 

Clippy generally has good advice

# What changes are included in this PR?

This fixes a number of the clippy lints, with localised suppression in the cases where fixing would result in a breaking API change

# Are there any user-facing changes?

No

### CODE REVIEW

File: parquet/src/util/test_common/page_util.rs [hunk 1]
```diff
@@ -229,88 +223,3 @@ impl<I: Iterator<Item = Vec<Page>> + Send> PageIterator for InMemoryPageIterator
         Ok(self.column_desc.clone())
     }
 }
-
-pub fn make_pages<T: DataType>(
```
Comment (tustvold) [inline parquet/src/util/test_common/page_util.rs:233]: This is moved to rand_gen, as it made the conditional compilation easier

File: parquet/src/encodings/levels.rs [hunk 2]
```diff
@@ -142,12 +142,14 @@ impl LevelEncoder {
 /// Decoder for definition/repetition levels.
 /// Currently only supports RLE and BIT_PACKED encoding for Data Page v1 and
 /// RLE for Data Page v2.
+#[allow(unused)]
```
Comment (tustvold) [inline parquet/src/encodings/levels.rs:145]: Interestingly this is never actually used anywhere :sweat_smile:

File: parquet/src/arrow/array_reader/map_array.rs [hunk 3]
```diff
@@ -41,14 +40,12 @@ impl MapArrayReader {
...
         data_type: ArrowType,
         def_level: i16,
-        rep_level: i16,
     ) -> Self {
         Self {
             key_reader,
             value_reader,
             data_type,
-            map_def_level: rep_level,
-            map_rep_level: def_level,
```
Comment (tustvold) [inline parquet/src/arrow/array_reader/map_array.rs:51]: I don't think this is correct, but changing it caused the tests to fail...

**Review by viirya — APPROVED** [review]
This only fixes lint errors. lgtm

**Pull request merged**